File: ParadigmasProgramacao/Tafera_4/PROGII/soma_media_prod_maior_menor.c
#include <stdio.h>
#include <stdlib.h>

/*
Escreva um programa que leia 3 inteiros e escreva a soma, média, produto, maior e menor desses números.
*/

void soma_media_prod_maior_menor()
{
    int numero1, numero2,numero3;
    int maior = -1, x;


    printf("Digite 3 números: \n");
    scanf("%d%d%d", &numero1, &numero2,&numero3);

    /*x = numero1 > numero2;

    printf("%x", x);

    if(2 > 1 > 0)
    {
        maior = numero1;
    }*/

    if(numero1 > numero2 > numero3)
    {
        maior = numero1;
    }

    if(numero2 > numero1 > numero3)
    {
        maior = numero1;
    }

    if(numero3 > numero2 > numero3)
    {
        maior = numero1;
    }

    printf("%d é o maior.\n", maior);



    // --//--

    printf("%d\n", (numero1+ numero2 + numero3) );

    printf("%d\n", (numero1 * numero2 * numero3) );

    printf("%f\n", ((float)(numero1+ numero2 + numero3) / 3) );

}

File: ParadigmasProgramacao/Tafera_4/SomaMediaProdMaiorMenor.py
class SomaMediaProdMaiorMenor:
    def __init__(self) -> None:
        print("Digite 3 números: ")
        self.n1 = float(input())
        self.n2 = float(input())
        self.n3 = float(input())
    
    def Maior(self):
        print(f"O Maior Valor é: {max([self.n1, self.n2, self.n3])}")

    def Menor(self):
        print(f"O Menor valor é: {min([self.n1, self.n2, self.n3])}")

    def Soma(self):
        print(f"A Soma é: {self.n1 + self.n2 + self.n3}")
    
    def Multiplicacao(self):
        print(f"A multiplicação é: {self.n1 * self.n2 * self.n3}")
    
    def Produto(self):
        print(f"O Produto é: {(self.n1 + self.n2 + self.n3)/2}")

if __name__ == "__main__":
    obj = SomaMediaProdMaiorMenor()
    obj.Maior()
    obj.Menor()
    obj.Soma()
    obj.Multiplicacao()
    obj.Produto()